find-phantom-resource.ps1

```
@@ -27,43 +27,272 @@
+    [switch]$DeleteResourceGroup
+)
+
+# Validate parameter combinations
+if ($Delete -and -not $Force) {
+    Write-Host "ERROR: -Delete requires -Force to confirm destructive operations." -ForegroundColor Red
+    Write-Host "  Use -DryRun to preview what would be deleted." -ForegroundColor Yellow
+    Write-Host "  Use -Delete -Force to actually delete." -ForegroundColor Yellow
+    exit 1
+}
+if ($DeleteResourceGroup -and -not $Delete) {
+    Write-Host "ERROR: -DeleteResourceGroup requires -Delete -Force." -ForegroundColor Red
```

PSScriptAnalyzer (1.24.0)

[warning] File 'find-phantom-resource.ps1' uses Write-Host. Avoid using Write-Host because it might not work in all hosts, does not work when there is no host, and (prior to PS 5.0) cannot be suppressed, captured, or redirected. Instead, use Write-Output, Write-Verbose, or Write-Information.

(PSAvoidUsingWriteHost)